lib/rubocop/cop/style/file_touch.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
```

Comment:
I'm wondering if this should be a Lint cop instead. [In the wild](https://github.com/search?q=%2Ffile%5C.open.%2B%5Cba%5C%2B%3F.%5C%29+%5C%7B%5C%7D%2F+language%3ARuby+&type=code&p=1), I've seen `File.open` used to make sure a file exists or [update its timestamps](https://github.com/joeyates/imap-backup/blob/ff09698d25e7033b1a2d8c182e95ed330572772a/lib/imap/backup/serializer/mbox.rb#L109-L113) (much like [`touch`](https://www.gnu.org/software/coreutils/manual/html_node/touch-invocation.html) utility), the latter of which actually doesn't happen:
```ruby
FileUtils.touch('foo.tmp') # create a file

atime_1, ctime_1, mtime_1 = File.atime('foo.tmp'), File.ctime('foo.tmp'), File.mtime('foo.tmp')

File.open('foo.tmp', 'a') {}

atime_2, ctime_2, mtime_2 = File.atime('foo.tmp'), File.ctime('foo.tmp'), File.mtime('foo.tmp')

atime_1 == atime_2 && ctime_1 == ctime_2 && mtime_1 == mtime_2
# => true
```
so this could perhaps be considered erroneous code.